samples/LibVLCSharp.Avalonia.Sample/ViewModels/MainWindowViewModel.cs

```diff
@@ -1,24 +1,37 @@
 ﻿using System;
+using Avalonia.Controls;
 using LibVLCSharp.Shared;
 
 namespace LibVLCSharp.Avalonia.Sample.ViewModels
 {
     public class MainWindowViewModel : ViewModelBase, IDisposable
     {
         private readonly LibVLC _libVlc = new LibVLC();
+        
+        public MediaPlayer MediaPlayer { get; }
+        
         public MainWindowViewModel()
         {
             MediaPlayer = new MediaPlayer(_libVlc);
+            Play();
```

Comment:
I'd avoid putting this kind of logic in a constructor (it does not initialize the properties of the object). Isn't there a "Loaded" event that you can leverage on the Control to trigger the Play method ?

Reply:
Maybe we could utilize **Initialized** event in this case:

> Occurs when the styled element has finished initialization.

> The Initialized event indicates that all property values on the styled element have been set. When loading the styled element from markup, it occurs when ISupportInitialize.EndInit is called *and* the styled element is attached to a rooted logical tree. When the styled element is created by code and ISupportInitialize is not used, it is called when the styled element is attached to the visual tree.

https://reference.avaloniaui.net/api/Avalonia/StyledElement/D1745B44

Reply:
yes, please :)

Reply:
Turns out that the Initialized event is fired before **Datacontext** is assigned to the control, so I think we could go with **DataContextChanged** event.  Video starts playing when viewmodel is assigned to usercontrol.

Reply:
Pushed changes.